xbmc/utils/StringUtils.cpp

```diff
@@ -512,6 +512,17 @@ std::string& StringUtils::TrimRight(std::string &str, const char* const chars)
   return str;
 }
 
+int StringUtils::ReturnDigits(std::string str)
+{
+  std::stringstream ss;
+  for (const auto& character : str)
+  {
+    if (isdigit(character))
+      ss << character;
+  }
+  return atoi(ss.str().c_str());
+}
+
```

Comment:
Not sure what you're trying to do here. It extracts all integers from the string, then return 0 on error anyway. Simply `atoi` should be sufficient, right?